setup.py

```diff
@@ -30,12 +32,26 @@ def main() -> None:
     """Builds python package used by Google Cloud Dataflow workers."""
     # Configure the required packages and scripts to install.
     with open('requirements.txt', encoding='utf-8') as requirements_txt: # pylint: disable=replace-disallowed-function-calls
+        requirements_content = requirements_txt.read()
+        # Removing the hashes from the requirements.txt file because they are
+        # not supported by the 'pkg_resources.parse_requirements' function while
+        # parsing the requirements.
+        modified_requirements_content = re.sub(
+            r'^\s*--hash=sha256:.*$|\\$',
+            '',
+            requirements_content,
+            flags=re.MULTILINE
+        )
+
         # The 'parse_requirements' returns a list of 'Requirement' objects.
         # We need to transform these to strings using the str() function.
         required_packages = [
             str(requirement)  # pylint: disable=replace-disallowed-function-calls
-            for requirement in pkg_resources.parse_requirements(
-                requirements_txt)
+            # TODO(#13528): Here we use MyPy ignore because we remove this test
+            # after the backend is fully type-annotated. Here ignore[arg-type]
+            # is used to test the constructor of ActivityReference for invalid
+            # argument type.
```

Comment:
I don't think this comment is correct. This isn't a test that will be removed after complete type annotation.

What type error are you getting without this ignore? From the `parse_requirements` code, it looks like a string is definitely an allowed input: https://github.com/pypa/setuptools/blob/6083e18f4afc40316c0112134c205c336afbcdfd/pkg_resources/__init__.py#L3158

Reply:
this was the error I was facing. PTAL
https://github.com/oppia/oppia/actions/runs/5556382341/jobs/10148824960

Reply:
For anyone reading this in the future, the error was `setup.py:48: error: Argument 1 to "parse_requirements" has incompatible type "str"; expected "TextIO"  [arg-type]`. This means that mypy thinks `pkg_resources.parse_requirements` only takes TextIO, even though its documentation clearly says that `str` is a valid type. Therefore, @shivkant5 I agree with adding a mypy ignore here like you already did. Please just update the comment to explain that the type hint on `pkg_resources.parse_requirements` is wrong

Reply:
changed the comment to have the actual prob. thanks!